src/EditorFeatures/CSharp/DecompiledSource/CSharpDecompiledSourceServiceFactory.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.DecompiledSource
+{
+    [ExportLanguageServiceFactory(typeof(IDecompiledSourceService), LanguageNames.CSharp), Shared]
+    internal partial class CSharpDecompiledSourceServiceFactory : ILanguageServiceFactory
+    {
+        public ILanguageService CreateLanguageService(HostLanguageServices provider)
```

Comment:
@sharwell has some patterns he'd like you do use here.  Sam, can you point him at the current pattern for mef types wrt Obsolete and whatnot?

Reply:
➡️ Will file as a follow-up issue.